src/XRP/model/xrp-account-set.ts

```diff
@@ -16,15 +17,44 @@ export default class XrpAccountSet {
    * @return an XrpAccountSet with its fields set via the analogous protobuf fields.
    * @see https://github.com/ripple/rippled/blob/3d86b49dae8173344b39deb75e53170a9b6c5284/src/ripple/proto/org/xrpl/rpc/v1/transaction.proto#L100
    */
-  public static from(accountSet: AccountSet): XrpAccountSet | undefined {
+  public static from(accountSet: AccountSet): XrpAccountSet {
     const clearFlag = accountSet.getClearFlag()?.getValue()
     const domain = accountSet.getDomain()?.getValue()
+    // domain must be lowercase
+    if (domain?.toLowerCase() != domain) {
+      throw new XrpError(
+        XrpErrorType.MalformedProtobuf,
+        'AccountSet protobuf field `domain` is not lowercase.',
+      )
+    }
     const emailHash = accountSet.getEmailHash()?.getValue_asU8()
     const messageKey = accountSet.getMessageKey()?.getValue_asU8()
     const setFlag = accountSet.getSetFlag()?.getValue()
     const transferRate = accountSet.getTransferRate()?.getValue()
+    // transferRate cannot be more than 2000000000 or less than 1000000000,
+    // except for the special case 0 meaning no fee.
+    if (transferRate) {
+      if (transferRate > 2000000000) {
+        throw new XrpError(
+          XrpErrorType.MalformedProtobuf,
+          'AccountSet protobuf field `transferRate` is above 2000000000.',
+        )
+      }
+      if (transferRate < 1000000000 && transferRate != 0) {
+        throw new XrpError(
+          XrpErrorType.MalformedProtobuf,
+          'AccountSet protobuf field `transferRate` is below 1000000000.',
+        )
+      }
+    }
     const tickSize = accountSet.getTickSize()?.getValue()
-
+    // Valid values for tickSize are 3 to 15 inclusive, or 0 to disable.
+    if (tickSize && (tickSize < 3 || tickSize > 15) && tickSize != 0) {
```

Comment:
can we either put these numbers in const or perhaps add a function called `isValidTickSize`?